src/qb4-packw/kr-neoni8mm.c.in

```diff
@@ -0,0 +1,193 @@
+// Copyright 2023 Google LLC
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+$ABC = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+$assert NR > 1
+$assert KR > 1
+#include <assert.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <math.h>
+#include <arm_neon.h>
+
+#include "src/xnnpack/packw.h"
+
+// convert a vector from packed nibbles to planar, and accumulate sum
+static XNN_INTRINSIC
+int8x16_t xnn_packed2planar(
+  int32x4_t *vacc,
+  const int8x16_t v,
+  const int8x16_t vmask,
+  const int8x16_t veor_mask,
+  const int32x4_t neg_zp,
+  const int8x16_t vones_zero_mask)
+{
+  const int8x16_t vl = vshrq_n_s8(v, 4);  // isolate lower int 4
+  const int8x16_t vh = vandq_s8(v, vmask);  // isolate upper int 4
+  *vacc = vmmlaq_s32(*vacc, vones_zero_mask, vl);
```

Comment:
so this kernel is i8mm because you sum up KR=8 using vmmlaq?
it is worth it?  You could do sdot and paired add outside loop

We've done neon c8 kernels before because i8mm is not really a win for m=1